File: DONGUK/프로그래머스_WEEK 2/짝지어 제거하기.py
from collections import deque

def solution(s):
    answer = -1
    
    if len(s) == 1:
        return 0
    
    q = []
    
    for i in range(len(s)):
        if i == 0:
            q.append(s[0])
        
        else:
            if len(q) == 0:
                q.append(s[i])
                continue
                
            if s[i] == q[-1]:
                q.pop()
            
            else:
                q.append(s[i])
                
    if q:
        answer = 0
    else:
        answer = 1
    return answer

File: JINSEOP/프로그래머스_지난_문제/짝지어 제거하기.cpp
#include <stack>
#include <string>
using namespace std;

stack<char> stacker;
void stack_and_erase(string s) {
    for (auto c : s) {
        if (stacker.empty())
            stacker.push(c);
        else {
            if (stacker.top() == c)
                stacker.pop();
            else
                stacker.push(c);
        }
    }
}

int solution(string s){
    int answer = -1;

    stack_and_erase(s);
    
    if (stacker.empty()) answer = 1;
    else answer = 0;

    return answer;
}
